applicationset/controllers/applicationset_controller.go

```diff
@@ -829,15 +849,18 @@ func (r *ApplicationSetReconciler) removeFinalizerOnInvalidDestination(ctx conte
 
 		// If the finalizer length changed (due to filtering out an Argo finalizer), update the finalizer list on the app
 		if len(newFinalizers) != len(app.Finalizers) {
+			updated := app.DeepCopy()
+			updated.Finalizers = newFinalizers
+			if err := r.Client.Patch(ctx, updated, client.MergeFrom(app)); err != nil {
+				return fmt.Errorf("error updating finalizers: %w", err)
+			}
+			if err := r.updateCache(ctx, updated); err != nil {
+				return fmt.Errorf("error updating app cache: %w", err)
+			}
 			app.Finalizers = newFinalizers
```

Comment:
This line is basically dead code now, right? Or does the caller reuse `app` with the new finalizers? I would assume that `updated` and `app` are now diverged, at least when it comes to resourceVersion.

Reply:
Unit test insisted finalizers need to be mutated on parameter. I looked at the code and suspect this is not required but scared to change my current behavior. You are right that  `updated` and `app` are now diverged - fixed by using `updated.DeepCopyInto(app)` to keep behavior as is. Thank you!